core/services/pipeline/task_params.go

```diff
@@ -0,0 +1,369 @@
+package pipeline
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/shopspring/decimal"
+	"github.com/smartcontractkit/chainlink/core/utils"
+	"go.uber.org/multierr"
+)
+
+type StringParam string
+
+func (s *StringParam) UnmarshalPipelineParam(val interface{}, vars Vars) error {
+	switch v := val.(type) {
+	case string:
+		*s = StringParam(v)
+		return nil
+	case []byte:
+		*s = StringParam(string(v))
+		return nil
+	default:
+		return ErrBadInput
+	}
+}
+
+type BytesParam string
+
+func (b *BytesParam) UnmarshalPipelineParam(val interface{}, vars Vars) error {
+	switch v := val.(type) {
+	case string:
+		if len(v) >= 2 && v[:2] == "0x" {
+			bs, err := hex.DecodeString(v[2:])
+			if err != nil {
+				return err
+			}
+			*b = BytesParam(bs)
+			return nil
+		}
+		*b = BytesParam(v)
+	case []byte:
+		*b = BytesParam(v)
+	default:
+		return ErrBadInput
+	}
+	return nil
+}
+
+type Uint64Param uint64
+
+func (u *Uint64Param) UnmarshalPipelineParam(val interface{}, vars Vars) error {
+	switch v := val.(type) {
+	case uint:
+		*u = Uint64Param(v)
+	case uint8:
+		*u = Uint64Param(v)
+	case uint16:
+		*u = Uint64Param(v)
+	case uint32:
+		*u = Uint64Param(v)
+	case uint64:
+		*u = Uint64Param(v)
+	case int:
+		*u = Uint64Param(v)
+	case int8:
+		*u = Uint64Param(v)
+	case int16:
+		*u = Uint64Param(v)
+	case int32:
+		*u = Uint64Param(v)
+	case int64:
+		*u = Uint64Param(v)
+	case string:
+		n, err := strconv.ParseUint(v, 10, 64)
+		if err != nil {
+			return errors.Wrap(ErrBadInput, err.Error())
+		}
+		*u = Uint64Param(n)
+		return nil
+	default:
+		return ErrBadInput
+	}
+	return nil
+}
+
+type MaybeUint64Param struct {
```

Comment:
Can we use `null.Uint64` instead of our own custom struct here?

Reply:
I don't think we import a `null` package anymore, and our own package only has Int64 and Uint32

Reply:
`guregu`'s `null` package doesn't appear to have a Uint64 type, only Int64